diff --git a/src/plugins/ctf/common/msg-iter/msg-iter.cpp b/src/plugins/ctf/common/msg-iter/msg-iter.cpp
index 5f38ec6c..354080de 100644
--- a/src/plugins/ctf/common/msg-iter/msg-iter.cpp
+++ b/src/plugins/ctf/common/msg-iter/msg-iter.cpp
@@ -1924,11 +1924,6 @@ static enum bt_bfcr_status bfcr_unsigned_int_char_cb(uint64_t value, struct ctf_
         goto end;
     }

-    if (value == 0) {
-        msg_it->done_filling_string = true;
-        goto end;
-    }
-
     string_field = stack_top(msg_it->stack)->base;
     BT_ASSERT_DBG(bt_field_get_class_type(string_field) == BT_FIELD_CLASS_TYPE_STRING);

