--- a/src/plugins/ctf/common/msg-iter/msg-iter.c	2021-02-10 00:13:50.634356354 +0000
+++ b/src/plugins/ctf/common/msg-iter/msg-iter.c	2021-02-10 00:14:14.990009271 +0000
@@ -2049,11 +2049,6 @@
 		goto end;
 	}
 
-	if (value == 0) {
-		msg_it->done_filling_string = true;
-		goto end;
-	}
-
 	string_field = stack_top(msg_it->stack)->base;
 	BT_ASSERT_DBG(bt_field_get_class_type(string_field) ==
 		BT_FIELD_CLASS_TYPE_STRING);
