diff --git a/src/common/gen-version-i.sh b/src/common/gen-version-i.sh
index 938d70c5..a8249d47 100755
--- a/src/common/gen-version-i.sh
+++ b/src/common/gen-version-i.sh
@@ -42,7 +42,7 @@ fi
 # configurations leading to different results.
 if test -r "$TOP_SRCDIR/bootstrap" && test -r "$TOP_SRCDIR/.git" &&
 		(command -v git > /dev/null 2>&1); then
-	GIT_VERSION_STR="$(cd "$TOP_SRCDIR" && git describe --tags --dirty --abbrev=12)"
+	GIT_VERSION_STR="$(cd "$TOP_SRCDIR" && git describe --tags --dirty --abbrev=12 --always)"
 	GIT_CURRENT_TAG="$(cd "$TOP_SRCDIR" && (git describe --tags --exact-match --match="v[0-9]*" HEAD || true) 2> /dev/null)"
 	echo "#define BT_VERSION_GIT \"$GIT_VERSION_STR\"" > version.i.tmp
 
