diff --git a/configure.ac b/configure.ac
index 2a78b946a..e653209a0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1054,6 +1054,12 @@ AS_IF([test "x$ARCHTYPE" = "xx86"],[
 ])
 AM_CONDITIONAL([HAVE_MODULES_USERSPACE_CALLSTACK_CONTEXT], [test x$have_modules_userspace_callstack_context != xno])
 
+AC_ARG_ENABLE([tests],
+  [AS_HELP_STRING([--enable-tests], [Build the tests. [default=yes]])],
+  [],
+  [enable_tests=yes]
+)
+
 # Export binaries build conditions.
 AM_CONDITIONAL([BUILD_BIN_LTTNG], [test x$enable_bin_lttng != xno])
 AM_CONDITIONAL([BUILD_BIN_LTTNG_CONSUMERD], [test x$enable_bin_lttng_consumerd != xno])
@@ -1067,12 +1073,22 @@ test "x$enable_bin_lttng" != "xno" && \
 test "x$enable_bin_lttng_consumerd" != "xno" && \
 test "x$enable_bin_lttng_crash" != "xno" && \
 test "x$enable_bin_lttng_relayd" != "xno" && \
-test "x$enable_bin_lttng_sessiond" != "xno"],
+test "x$enable_bin_lttng_sessiond" != "xno" && \
+test "x$enable_tests" == "xyes"],
 [build_tests=yes],
 [build_tests=no]
 )
 
 AM_CONDITIONAL([BUILD_TESTS], [test x$build_tests = xyes])
+AS_IF([test "x$build_tests" = "xyes"],
+  [
+    PKG_CHECK_MODULES([babeltrace2], [babeltrace2])
+    PKG_CHECK_MODULES([babeltrace2],[babeltrace2],,[
+    AC_MSG_ERROR([libbabeltrace2 is required to build tests])
+    ])
+  ]
+)
+
 AM_CONDITIONAL([BUILD_EXTRAS], [test x$enable_extras != xno])
 
 # Export libraries build conditions.
@@ -1322,6 +1338,10 @@ AS_IF([test "x$build_tests" = "xno"],[
 	PPRINT_WARN([Tests won't be built since some binaries were disabled])
 ])
 
+# Tests disable overall
+test "x$enable_tests" = "xyes" && value=1 || value=0
+PPRINT_PROP_BOOL([Tests], $value)
+
 # LTTng UST Java agent JUL tests enabled/disabled
 test "x$test_java_agent_jul" = "xyes" && value=1 || value=0
 PPRINT_PROP_BOOL([LTTng-UST Java agent JUL tests], $value)
